druid/src/env.rs

```diff
@@ -205,24 +215,30 @@ impl Env {
     ///
     /// Panics if the key is not found, or if it is present with the wrong type.
     pub fn get<'a, V: ValueType<'a>>(&'a self, key: impl Borrow<Key<V>>) -> V {
-        let key = key.borrow();
-        if let Some(value) = self.0.map.get(key.key) {
-            value.to_inner_unchecked()
-        } else {
-            panic!("key for {} not found", key.key)
+        match self.try_get(key) {
+            Ok(value) => value,
+            Err(err) => panic!("{}", err),
         }
     }
 
-    /// Gets a value from the environment.
+    /// Trys to get a value from the environment.
+    ///
+    /// If the value is not found, the raw key is returned as the error.
     ///
     /// # Panics
     ///
     /// Panics if the value for the key is found, but has the wrong type.
-    pub fn try_get<'a, V: ValueType<'a>>(&'a self, key: impl Borrow<Key<V>>) -> Option<V> {
+    pub fn try_get<'a, V: ValueType<'a>>(
+        &'a self,
+        key: impl Borrow<Key<V>>,
+    ) -> Result<V, MissingKeyError> {
         self.0
             .map
             .get(key.borrow().key)
             .map(|value| value.to_inner_unchecked())
+            .ok_or(MissingKeyError {
+                key: key.borrow().key,
```

Comment:
I'm having trouble seeing how this can work if the key is not in fact `&'static str`, as it seems `MissingKeyError` will constrain the lifetime. But if CI accepts it, I'm not going to complain.

Reply:
Good point, I think I'm counting on the fact that keys are currently always static, but that we'd like non-static ones to exist. I should have a test case for that...